datafusion/proto/src/physical_plan/to_proto.rs

```diff
@@ -891,6 +891,9 @@ impl TryFrom<&FileTypeWriterOptions> for protobuf::FileTypeWriterOptions {
             FileTypeWriterOptions::Arrow(ArrowWriterOptions {}) => {
                 return not_impl_err!("Arrow file sink protobuf serialization")
             }
+            FileTypeWriterOptions::Extension(_) => {
```

Comment:
We should be able to support serializing this since it is ultimately just a Vec<(String, String)>

Reply:
We could also punt it into the `FileType::Extension` (e.g. add a `serialize(&self) -> Result<Vec<u8>>` type method)